compose-builder/tui-generator.sh

```diff
@@ -0,0 +1,293 @@
+    [ds-virtual]="Include Virtual Device Service"
+    [ds-coap]="Include CoAP Device Service"
+    [ds-gpio]="Include GPIO Device Service"
+    [ds-llrp]="Include RFID LLRP Device Service"
+
+)
+
+
+## Available Message Bus Services in EdgeX-Foundry
+messageBus=(
+    "mqtt-bus"
+    "zmq-bus"
+)
+
+
+## Message Bus Descriptions
+declare -A msgBusDesc=(
+    [mqtt-bus]="Configure MQTT Message Bus"
+    [zmq-bus]="Configure ZMQ Message Bus"
+)
+
+
+####################################################################
+#    Additional Services Options Display Function                  #
+####################################################################
+function additionalServiceOption() {
+    message="Some Additional Services are available that can also be included. \n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip \n
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${additionalOpts[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${additionalOptsDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_OTHERS+=$(/usr/bin/whiptail --title "App Services" \
+                --notags --separate-output \
+                --ok-button Next \
+                --nocancel \
+                --checklist "$message" 20 75 5 \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+
+####################################################################
+#    App Services Options Display Function                         #
+####################################################################
+function appServiceOption() {
+    message="Available App Services to include in your compose file. \n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip \n
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${appServices[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${appServiceDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_APPSERVICES+=$(/usr/bin/whiptail --title "App Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --checklist "$message" 20 75 7 \
+                -- "${arglist[@]}" \
+                3>&1 1>&2 2>&3)
+}
+
+
+
+####################################################################
+#    ARM64 Question Display Function                               #
+####################################################################
+function displayArm64() {
+    message="Would you like to use ARM64 Images to generate the Compose file?"
+
+    /usr/bin/whiptail --title "Images Architecture" --yesno "$message" 8 78
```

Comment:
Reverse the logic here since `Not ARM64` will be the most common answer so should be the default.

Reply:
This is resolved by setting `--defaultno` on the Yes / No Message box. By default the box will have the "cursor" set on No.